src/datasources/entity-ds/datasource.js

```diff
@@ -361,7 +361,14 @@ export class OpenNMSEntityDatasource {
     }
 
     clearAlarm(alarmId) {
-        return this.opennmsClient.doClear(alarmId, this.user);
+        const clearAlarmDefer = this.q.defer();
+        this.q.when(this.opennmsClient.doClear(alarmId, this.user)).then(() => {
+            clearAlarmDefer.resolve({status: true})
+        }, (error) => {
+            console.log("DataSource Clear Alarm Error: ", error)
+            clearAlarmDefer.reject({status: false, error: error})
+        });
+        return clearAlarmDefer.promise;
```

Comment:
`doClear` is already a non-angularjs Promise; I think you should be able to just do this.q.when() to wrap it in an angular Promise without having to do the extra `defer` stuff.  (`q.when` will auto-convert any "then-able" promise to a Q promise.)  If you still want to return a "status" object you should be able to do this:

```javascript
return this.q.when(this.opennmsClient.doClear(alarmId, this.user)).then(() => {
  return { status: true };
}).catch((err) => {
  console.log("DataSource Clear Alarm Error:", error);
  return { status: false, error: error };
});
```

Reply:
Thats right! I got it now. Thanks for mentioning! Updated the pull request for this in commit: https://github.com/OpenNMS/opennms-helm/pull/254/commits/46fb262bc1269eb2659846ff59f9aafaa8d482ab